virtcontainers/qemu.go

```diff
@@ -646,17 +646,19 @@ func (q *qemu) stopSandbox() error {
 
 	err := q.qmpSetup()
 	if err != nil {
-		return err
+		q.Logger().WithError(err).Error("Failed to setup qmp")
```

Comment:
Well this is a real error, so putting a warning here does not seem right.

@nitkon I think you should put the whole VM cleanup in its own function so that it can be called from a `defer` statement before we try to call into `qmpSetup()`. The idea would be to cleanup the VM in every case, no matter if there's an error or not. So you should do something like:
```go
defer cleanupVM()
...
if err := qmpSetup(); err != nil {
   return err
}
...
qmp.ExecuteQuit()
...
return nil
```
with the function `cleanupVM()` taking care of checking if the VM is still up, and if that's the case, `SIGKILL` it. We need to check if the PID of the VM is stored somewhere or if that's an info we can get.
Of course, `cleanupVM()` would also remove the VM paths and dirs.

Reply:
It can be **dangerous** to `SIGKILL` the qemu process at the point. When qmp connection fails, most likely the qemu process have crashed. If we just send `SIGKILL` to its pid, we might end up killing some other process mistakenly.

Reply:
Good point, but then we won't do anything about a Qemu process that might be in a weird state.

Reply:
Either we leave it as is, or we verify the qemu process cmd args against `q.state.UUID` before sending `SIGKILL` to it.

Reply:
That could be a way, yes. But I think for now this PR wants to focus on cleaning up files, so let's keep it simple.
@nitkon could you not remove the `return err` in both `qmpSetup()` and `ExecuteQuit()` functions, but instead use the `defer` approach. This is cleaner since we understand by reading the code that the defer statement will be run no matter if there is an error or not.